src/editor.js

```diff
@@ -99,61 +98,43 @@ class Editor {
     $initOperationListeners() {
         this.commands.on("exec", this.startOperation.bind(this), true);
         this.commands.on("afterExec", this.endOperation.bind(this), true);
-
-        this.$opResetTimer = lang.delayedCall(this.endOperation.bind(this, true));
-        
-        // todo: add before change events?
-        this.on("change", function() {
-            if (!this.curOp) {
-                this.startOperation();
-                this.curOp.selectionBefore = this.$lastSel;
-            }
-            this.curOp.docChanged = true;
-        }.bind(this), true);
-        
-        this.on("changeSelection", function() {
-            if (!this.curOp) {
-                this.startOperation();
-                this.curOp.selectionBefore = this.$lastSel;
-            }
-            this.curOp.selectionChanged = true;
-        }.bind(this), true);
     }
 
     startOperation(commandEvent) {
-        if (this.curOp) {
-            if (!commandEvent || this.curOp.command)
-                return;
-            this.prevOp = this.curOp;
-        }
+        this.session.startOperation(commandEvent);
+    }
+
+    /** 
+     * @arg e
+     */
+    endOperation(e) {
+        this.session.endOperation(e);
+    }
+
+    onStartOperation(commandEvent) {
+        // scrollTop is kept inside of session.curOp only for backwards compatibility reasons
```

Comment:
why?

Reply:
Just because the bottom line `this.curOp = this.session.curOp = ...` was inserting it. 
So someone could had have integration code doing something like `session.curOp.scrollTop`. I fully agree though that conceptually this shouldn't be there...   

Regarding backwards compatibility, are you okay with just treating this as an implementation detail on our end, since we don't really officially document it in https://github.com/ajaxorg/ace/blob/37c5f7602b33dd05bba5c2521abbfe7bb2cc651a/ace.d.ts#L507 or in https://ajaxorg.github.io/ace-api-docs/index.html?